api/v1/handlers.go

```diff
@@ -1758,35 +1758,62 @@ func (s *apiV1) handleGetSystemConfig(c echo.Context, u *util.User) error {
 }
 
 type minerResp struct {
-	Addr            address.Address `json:"addr"`
-	Name            string          `json:"name"`
-	Suspended       bool            `json:"suspended"`
-	SuspendedReason string          `json:"suspendedReason,omitempty"`
-	Version         string          `json:"version"`
+	Addr            address.Address       `json:"addr"`
+	Name            string                `json:"name"`
+	Suspended       bool                  `json:"suspended"`
+	SuspendedReason string                `json:"suspendedReason,omitempty"`
+	Version         string                `json:"version"`
+	ChainInfo       *miner.MinerChainInfo `json:"chain_info"`
 }
 
 // handleAdminGetMiners godoc
 // @Summary      Get all miners
-// @Description  This endpoint returns all miners
-// @Tags         public,net
+// @Description  This endpoint returns all miners. Note: value may be cached
+// @Tags         admin,net
 // @Produce      json
 // @Success      200  {object}  string
 // @Failure      400           {object}  util.HttpError
 // @Failure      500           {object}  util.HttpError
-// @Router       /public/miners [get]
+// @Router       /admin/miners/ [get]
 func (s *apiV1) handleAdminGetMiners(c echo.Context) error {
+	ctx := context.TODO()
+
 	var miners []model.StorageMiner
 	if err := s.DB.Find(&miners).Error; err != nil {
 		return err
 	}
 
 	out := make([]minerResp, len(miners))
+
 	for i, m := range miners {
 		out[i].Addr = m.Address.Addr
 		out[i].Suspended = m.Suspended
 		out[i].SuspendedReason = m.SuspendedReason
 		out[i].Name = m.Name
 		out[i].Version = m.Version
+
+		var ci *miner.MinerChainInfo
+
+		// Cache the Chain Lookup for this miner, looking up if it doesnt exist / is expired
+		key := cacheKey(c, nil, m.Address.Addr.String())
+		cached, ok := s.extendedCacher.Get(key)
+		if ok {
+			cachedCi, ok := cached.(*miner.MinerChainInfo)
+			if ok {
+				ci = cachedCi
+			} else {
+				newCi, err := s.minerManager.GetMinerChainInfo(ctx, m.Address.Addr)
```

Comment:
this through adding to the cache should happen outside of the top-level "if ok" block